x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1119,11 +1119,8 @@ public AsyncSender interceptSender(AsyncSender sender) {
 
     @Override
     public UnaryOperator<RestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
-        if (enabled == false) {
-            return null;
-        }
-        final boolean ssl = HTTP_SSL_ENABLED.get(settings);
-        final SSLConfiguration httpSSLConfig = getSslService().getHttpTransportSSLConfiguration();
+        final boolean ssl = enabled && HTTP_SSL_ENABLED.get(settings);
+        final SSLConfiguration httpSSLConfig = ssl ? getSslService().getHttpTransportSSLConfiguration() : null;
         boolean extractClientCertificate = ssl && getSslService().isSSLClientAuthEnabled(httpSSLConfig);
```

Comment:
I think it reads better if we inline `httpSSLConfig`. Though it works in this case, assigning a variable to `null` and then deciding whether to use it based on another condition bother me.

Reply:
Sorry but what I meant is to replace
```java
final SSLConfiguration httpSSLConfig = ssl ? getSslService().getHttpTransportSSLConfiguration() : null;        
boolean extractClientCertificate = ssl && getSslService().isSSLClientAuthEnabled(httpSSLConfig);
```

with 

```java
boolean extractClientCertificate = 
    ssl && getSslService().isSSLClientAuthEnabled(getSslService().getHttpTransportSSLConfiguration());
```